src/transformers/integrations/awq.py

```diff
@@ -372,3 +386,18 @@ def _fuse_awq_attention_layers(model, module, modules_to_fuse, current_module_na
         setattr(parent, child_name, fused_attention_layer.to(previous_device))
 
         del q_proj, k_proj, v_proj, o_proj
+
+
+def post_init_awq_exllama_modules(model):
+    """
+    Runs post init for Exllama layers which performs:
+        - Weights unpacking, reordering and repacking
+        - Devices scratch space allocation
+    """
+    from awq.modules.linear.exllamav2 import exllamav2_post_init
+
+    # default values for exllamav2 from
+    # https://github.com/AutoGPTQ/AutoGPTQ/blob/6ba14f17ef73c161c2c4707cbf0b41e569a9c6dd/auto_gptq/nn_modules/qlinear/qlinear_exllamav2.py#L171
+    model = exllamav2_post_init(model, max_input_len=2048, max_batch_size=8)
```

Comment:
couldn't we make `max_input_len` configurable through `AwqConfig` - wdyt?

Reply:
Marc suggested we leave it as is for now https://github.com/huggingface/transformers/pull/28634#discussion_r1463506812

Reply:
oh okay ! I think it would makes sense to directly expose a `exllama_config` I think - wdyt @SunMarc ?

Reply:
I guess in another PR right ?

Reply:
Hmmm I think it should be better to add it now and not leave the main branch with hardcoded config values, it shouldn't be super complex as you can just copy over the existing logic in GptqConfig right?